chiselFrontend/src/main/scala/Chisel/Aggregate.scala

```diff
@@ -147,7 +157,7 @@ sealed class Vec[T <: Data] private (gen: => T, val length: Int)
   def read(idx: UInt): T = apply(idx)
 
   @deprecated("Use Vec.apply instead", "chisel3")
-  def write(idx: UInt, data: T): Unit = apply(idx) := data
+  def write(idx: UInt, data: T): Unit = (apply(idx) := data)(DeprecatedSourceInfo())
```

Comment:
Please use the . syntax for operations when you do this:
apply(idx).:=(data(DeprecatedSourceInfo()))

Although neither looks great.